exec-tests

```diff
@@ -140,16 +140,22 @@ _cov_report="${_cov_report_kind}:${_cov_report_name}"
 rc=0
 
 if [[ "${subtst:-python}" == "python" ]]; then
-    _pytest_majors="pbench.test.unit.common"
-    for _major in ${major_list}; do
-        _pytest_majors="${_pytest_majors} pbench.test.unit.${_major}"
-        if [[ "${_major}" == "agent" ]]; then
-            # TODO: real functional tests require a deployed instance. Current
-            # agent "functional" tests are mocked Click tests rather than true
-            # "functional" tests.
-            _pytest_majors="${_pytest_majors} pbench.test.functional.agent"
-        fi
+    no_tests_specified=1
+    for word in ${posargs}; do
+        if [[ ${word} != -* ]]; then no_tests_specified=0; fi
```

Comment:
I actually went back and forth when I was writing this.  I settled on the above approach because of the _next_ line (147):
```
    if [[ ${no_tests_specified} -eq 1 ]]; then
```

Does this read better?
```
    tests_specified=0
    for word in ${posargs}; do
        if [[ ${word} != -* ]]; then tests_specified=1; fi
    done
    if [[ ${tests_specified} -eq 0 ]]; then
```
The normal case is that there are no tests specified, so I set up the code to make that be the "true" value.  (I think it would read better without the `-eq X`, but I'm not sure how to do that in Bash....)

Reply:
Yeah, I agree nothing in bash is ideal, but I like `tests_specified` better than `no_tests_specified`. Is it worth another commit and review pass? Well, I dunno.

Reply:
According to https://docs.pytest.org/en/7.1.x/example/pythoncollection.html#interpreting-cmdline-arguments-as-python-packages, `--pyargs` is not for options but for arguments, what tests to run and how to interpret those arguments for test discovery.  We can move `--pyargs` into the `pytest.ini` file if that helps.

Reply:
OK, so long as `--pyargs` is a stand-alone option (i.e., it affects how PyTest interprets command line arguments (and not options)) and doesn't take any argument-value itself, then there's no dilemma.  😅 

Moving `--pyargs` to `pytest.ini` sounds like a good idea, but I don't think I'll add that to this PR.

And, I guess I'll merge this as is, then.  (We can always change it later....)